golangci-latest job fails with typecheck errors due to missing config.toml

The `golangci-latest` CI job is currently failing with `typecheck` errors when running golangci-lint:

```
pkg/config/config.go:250:12: pattern config.toml: no matching files found (typecheck)
//go:embed config.toml
```

This happens because `pkg/config/config.toml` is generated in the `config` job but is not available in the `golangci-latest` job. As a result, the linter cannot build the package, and `only-new-issues` filtering does not apply.

**Steps to Reproduce:**

* Open a pull request that does not touch `pkg/config/config.go` or `pkg/config/config.toml`.
* Observe that `golangci-latest` fails with a `typecheck` error.

```
run golangci-lint
  only new issues on pull_request: /tmp/tmp-2138-p9VgJ0wsaW17/pull.patch
  Running [/home/runner/golangci-lint-2.5.0-linux-amd64/golangci-lint config path] in [/home/runner/work/cli/cli] ...
  Running [/home/runner/golangci-lint-2.5.0-linux-amd64/golangci-lint config verify] in [/home/runner/work/cli/cli] ...
  Running [/home/runner/golangci-lint-2.5.0-linux-amd64/golangci-lint run --new-from-patch=/tmp/tmp-2138-p9VgJ0wsaW17/pull.patch --new=false --new-from-rev= --new-from-merge-base=] in [/home/runner/work/cli/cli] ...
  Error: pkg/config/config.go:250:12: pattern config.toml: no matching files found (typecheck)
  //go:embed config.toml
             ^
  1 issues:
  * typecheck: 1
  
  Error: issues found
  Ran golangci-lint in 21267ms
```

**Expected Behavior:**

The `golangci-latest` job should run successfully and only report linter issues introduced by the PR diff.

**Proposed Fix:**

* Make `golangci-latest` depend on the config job.
* Download the `pkg/config/config.toml` artifact into the correct path before running `golangci-lint`.
* Optionally verify the file exists to avoid typecheck failures.

--- a/.github/workflows/pr_test.yml
+++ b/.github/workflows/pr_test.yml
@@ -127,11 +127,22 @@ jobs:
           for dockerFile in Dockerfile*; do docker build -f $dockerFile . ; done
   golangci-latest:
     name: lint-latest (informational)
+    needs: [config]
     runs-on: ubuntu-latest
     continue-on-error: true
     steps:
       - name: Checkout code
         uses: actions/checkout@v5
+        with:
+          fetch-depth: 0
+      - name: Download config artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: config-artifact-${{ github.sha }}
+          path: pkg/config
+      - name: Verify embedded config exists
+        run: |
+          test -f pkg/config/config.toml || { echo "missing pkg/config/config.toml"; ls -la pkg/config; exit 1; }
       - name: Setup Go
         uses: actions/setup-go@v6
         with:

